src/processor.cpp

```diff
@@ -0,0 +1,413 @@
+        optional( messages_obj, was_loaded, "start", start_message, translated_string_reader );
+        optional( messages_obj, was_loaded, "processing", processing_message, translated_string_reader );
+        optional( messages_obj, was_loaded, "finish", finish_message, translated_string_reader );
+        optional( messages_obj, was_loaded, "fail", fail_message, translated_string_reader );
+    }
+}
+
+void processor_data::check() const
+{
+    static const iexamine_function converter( iexamine_function_from_string( "converter" ) );
+    const furn_str_id processor_id( id.str() );
+
+    if( !processor_id.is_valid() ) {
+        debugmsg( "Furniture \"%s\" have no processor of the same name.", id.c_str() );
+    } else if( processor_id->examine != converter ) {
+        debugmsg( "Furniture \"%s\" can't act as a processor because the assigned iexamine function is not converter, but it has \"%s\" associated.",
+                  processor_id.c_str(), id.c_str() );
+    }
+    for( const process_id &elem : processes ) {
+        if( !elem.is_valid() ) {
+            debugmsg( "Invalid process \"%s\" in \"%s\".", elem.c_str(), id.c_str() );
+        }
+    }
+}
+
+void processor_data::load( JsonObject &jo, const std::string &src )
+{
+    processors_data.load( jo, src );
+}
+
+void processor_data::check() const
+{
+    processors_data.check();
+}
+
+void processor_data::reset()
+{
+    processors_data.reset();
+}
+
+//process namespace
+
+namespace
+{
+struct process_data;
+using process_id = string_id<process_data>;
+
+struct process_data {
+
+    process_data() :
+        was_loaded(false) {};
+
+    process_id id;
+    std::string name;
+    std::vector<itype_id> components;
+    itype_id output;
+    int fuel_intake;
+    int duration;
+    //TODO: Skill used
+
+    bool was_loaded;
+    void load( JsonObject &jo, const std::string &src );
+    void check() const;
+    void reset();
+
+};
+
+process_id get_process_id( std::string id )
+{
+    return process_id( id );
+}
+
+generic_factory<process_data> processes_data( "process type", "id" );
+
+}
+
+void process_data::load( JsonObject &jo, const std::string & )
+{
+    mandatory( jo, was_loaded, "name", name );
+    //TODO: modify to allow lists of components
+    mandatory( jo, was_loaded, "components", components,  string_id_reader<itype_id> {} );
+    mandatory( jo, was_loaded, "output", output, string_id_reader<itype_id> {} );
+    mandatory( jo, was_loaded, "fuel_intake", fuel_intake );
+    mandatory( jo, was_loaded, "duration", duration );
+}
+
+void process_data::check() const
+{
+    const process_id proc_id( id.str() );
+
+    if (!proc_id.is_valid()) {
+        debugmsg("There is no process with the name %s", id.c_str());
+    }
+    /*
+    I'm not sure if I need this, it would validate components and output as valid itype_id
+    for( itype_id elem : components ) {
+        //The compiler complains about this is_valid method
+        if( !elem.is_valid() ) {
```

Comment:
That's because `itype_id` is not an id, despite the name. It's a deficiency of our code, not your fault.
You can check with `item::type_is_defined( id_here )`.